iris-mpc-store/src/lib.rs

```diff
@@ -566,6 +567,88 @@ WHERE id = $1;
         Ok(())
     }
 
+    /// Retrieve entry from `persistent_state` table with associated `domain` and `key` identifiers.
+    ///
+    /// Returns `Some(value)` if an entry exists for these identifiers and deserialization to generic
+    /// type `T` succeeds.  If no entry exists, then returns `None`.
+    pub async fn get_persistent_state<T: DeserializeOwned>(
+        &self,
+        domain: &str,
+        key: &str,
+    ) -> Result<Option<T>> {
+        let row = sqlx::query(
+            r#"
+            SELECT "value"
+            FROM persistent_state
+            WHERE domain = $1 AND "key" = $2
+            "#,
+        )
+        .bind(domain)
+        .bind(key)
+        .fetch_optional(&self.pool)
+        .await?;
+
+        if let Some(row) = row {
+            let value = row.try_get("value")?;
+            let deserialized: T = serde_json::from_value(value)?;
+            Ok(Some(deserialized))
+        } else {
+            Ok(None)
+        }
+    }
+
+    /// Set the entry in the `persistent_state` table for primary key `(domain, key)`.
+    ///
+    /// If an entry already exists for this primary key, this overwrites the existing
+    /// value with the value specified here.
+    pub async fn update_persistent_state<T: Serialize>(
```

Comment:
```suggestion
    pub async fn upsert_persistent_state<T: Serialize>(
```
@bgillesp which name do you prefer?

Reply:
Good point that `update` implies an existing entry is being modified.  `upsert` is quite correct in terms of functionality but I fear it's a bit less legible -- could we just go with something simple like `set_persistent_state`?

Reply:
Yes. Done.